pkg/system/phase2_creating.go

```diff
@@ -408,6 +420,12 @@ func (r *Reconciler) setDesiredCoreEnv(c *corev1.Container) {
 func (r *Reconciler) SetDesiredCoreApp() error {
 	r.CoreApp.Spec.Template.Labels["noobaa-core"] = r.Request.Name
 	r.CoreApp.Spec.Template.Labels["noobaa-mgmt"] = r.Request.Name
+	if coreLabels, ok := r.NooBaa.Spec.Labels["core"]; ok {
```

Comment:
I understand that the intention is to take the labels and annotations from the CR spec.
But do we actually want to override the previously assigned labels like system name (two lines above), isn't our intention to merge?
If that is not the case then I do not understand why would we assign labels and then override them completely in the next line.
This would break any pre-existing code that looked up for `noobaa-core` and `noobaa-mgmt` labels.

Reply:
good catch, sorry I messed up the order during all the rebase.